application/ui/src/features/annotator/annotations/annotation-shape-with-labels.component.tsx

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+import polylabel from 'polylabel';
+import { useSecondaryToolbarState } from 'src/features/dataset/media-preview/secondary-toolbar/use-secondary-toolbar-state.hook';
+
+import { Annotation, Polygon } from '../types';
+import { AnnotationLabels } from './annotation-labels.component';
+import { getFormattedPoints } from './utils';
+
+type AnnotationShapeProps = {
+    annotation: Annotation;
+};
+export const AnnotationShapeWithLabels = ({ annotation }: AnnotationShapeProps) => {
+    const { shape, labels } = annotation;
+    const color = labels.length ? labels[0].color : '--annotation-fill';
+    const { removeLabels } = useSecondaryToolbarState();
+
+    if (shape.type === 'rectangle') {
+        return (
+            <g transform={`translate(${shape.x}, ${shape.y})`}>
+                <rect aria-label='annotation rect' x={0} y={0} width={shape.width} height={shape.height} fill={color} />
+                <AnnotationLabels labels={labels} onRemove={removeLabels} />
```

Comment:
In Geti when the user removes the label we remove the shape as well. I think this would make sense here as well.

If a user would want to remove the label without removing the shape, then that would actually mean that they want to change the label into another label. We can implement this the same way that we did with Geti: show the label tree picker that allows the user to add / remove labels the the annotation.

Reply:
The user can have multiple labels and only wanting to remove one. We can obviously check the #labels but I think shape being separate from labels is a plus

Reply:
> The user can have multiple labels and only wanting to remove one

That's why the label picker is very useful. They can click the picker and we show a tree where when you hover over the labels it gives a hint on if the label gets added or removed.

Reply:
Merging this right now because it's blocking, but we can iterate on this.